src/inc/Lfo.h

```diff
@@ -0,0 +1,88 @@
+#if !defined(__Lfo_hdr__)
+#define __Lfo_hdr__
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+#include "ErrorDef.h"
+#include "Synthesis.h"
+#include "RingBuffer.h"
+
+class CLfo
+{
+public:
+    CLfo(float fModFreq, float fSampleRate)
+    {
+        m_pCRingBuffer = 0;
+        m_fRead = 0;
+        m_fModFreq = fModFreq;
+        m_fSampleRate = fSampleRate;
+        if (m_fModFreq == 0) {
+            m_iBufferLength = 1;
+        } else {
+            m_iBufferLength = static_cast<int>(1 / m_fModFreq);
+        }
+        m_pCRingBuffer = new CRingBuffer<float>(m_iBufferLength);
+    }
```

Comment:
Within this constructor is where I would have generated all the values for the wavetable and put them in `m_pCRingBuffer`. Then, have `currentIndex` and `indexIncrement` variables that represent your current position in and current frequency of the wavetable. This will allow you to simply call `m_fCRingBuffer.get(currentIndex)` within your `process()` to get the appropriate value

Reply:
I just renewed the function, now it looks pretty concise.
```
Lfo.h - Line 65
    Error_t generate () {
        CSynthesis::generateSine(m_pSine, m_fModFreqRatio * m_fSampleRate, m_fSampleRate, m_iWaveLength);
        m_pRingBuffer -> putPostInc(m_pSine, m_iWaveLength);
    }
    
    float getValue () {
        return m_pRingBuffer -> getPostInc();
    }
```